rust/src/client.rs

```diff
@@ -115,6 +115,10 @@ pub(crate) fn is_rpm_arg(arg: &str) -> bool {
     arg.ends_with(".rpm") || arg.starts_with("file://")
 }
 
+pub(crate) fn is_src_rpm_arg(arg: &str) -> bool {
+    arg.ends_with("src.rpm")
```

Comment:
```suggestion
    arg.ends_with(".src.rpm")
```

?

This probably works to start, though I think a more foolproof approach would be to actually check the RPM header for the `RPMTAG_SOURCEPACKAGE` flag. It gets set only on SRPMs:

```
$ rpm -q --qf 'SOURCEPACKAGE: %{SOURCEPACKAGE}\n' -p rpm-ostree-2021.14-2.fc34.src.rpm
SOURCEPACKAGE: 1
$ rpm -q --qf 'SOURCEPACKAGE: %{SOURCEPACKAGE}\n' -p rpm-ostree-2021.14-2.fc34.x86_64.rpm
SOURCEPACKAGE: (none)
```

Reply:
cool, I'll give this a go

Reply:
New approach seems to work, however the test becomes a bit more complicated I think. @cgwalters @jlebon any ideas on how to deal with the test? I bet we don't want to be pulling a src.rpm each time we run `make check`.
Leave the negative test only here and deal with an actual src.rpm in a kola test?